webhooky/tests/auth.rs

```diff
@@ -0,0 +1,741 @@
+}
+
+/// Test Checkr signatures
+
+#[ignore]
+#[tokio::test]
+async fn test_checkr_hmac_passes() {
+    let (port, _server) = make_server();
+
+    let test_signature = "66781e800f7d2934506890f5546af7736f1f84c46be507a7042f0be4e92259a0";
+    let test_body = include_str!("../tests/checkr_webhook_sig_test.json").trim();
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/checkr/verify", port))
+        .header("X-Checkr-Signature", test_signature)
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::ACCEPTED);
+}
+
+#[ignore]
+#[tokio::test]
+async fn test_checkr_hmac_fails() {
+    let (port, _server) = make_server();
+
+    let test_signature = "66781e800f7d2934506890f5546af7736f1f84c46be507a7042f0be4e92259b0";
+    let test_body = include_str!("../tests/checkr_webhook_sig_test.json").trim();
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/checkr/verify", port))
+        .header("X-Checkr-Signature", test_signature)
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::UNAUTHORIZED);
+}
+
+#[ignore]
+#[tokio::test]
+async fn test_checkr_hmac_audit_passes_with_invalid_signature() {
+    let (port, _server) = make_server();
+
+    let test_signature = "66781e800f7d2934506890f5546af7736f1f84c46be507a7042f0be4e92259b0";
+    let test_body = include_str!("../tests/checkr_webhook_sig_test.json").trim();
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/checkr/audit", port))
+        .header("X-Checkr-Signature", test_signature)
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::ACCEPTED);
+}
+
+/// Test DocuSign signatures
+
+#[tokio::test]
+async fn test_docusign_hmac_passes() {
+    let (port, _server) = make_server();
+
+    let test_signature = "048a1564644f631795724ec078399d672b09a254b3adaf84e4b20100e0564216";
+    let test_body = include_str!("../tests/docusign_webhook_sig_test.json").trim();
+
+    // Make the post API call.
+    let client = reqwest::Client::new();
+    let response = client
+        .post(format!("http://127.0.0.1:{}/hmac/docusign/verify", port))
+        .header("X-DocuSign-Signature-1", test_signature)
+        .body(test_body)
+        .send()
+        .await
+        .unwrap();
+
+    assert_eq!(response.status(), reqwest::StatusCode::ACCEPTED);
+}
+
+#[tokio::test]
+async fn test_docusign_hmac_fails() {
+    let (port, _server) = make_server();
+
+    let test_signature = "048a1564644f631795724ec078399d672b09a254b3adaf84e4b20100e0564217";
```

Comment:
Staring at this and the test above, I see that it's incremented the last digit by one. But it'd be immediately obvious that it should fail with something like `test_signature = 111111111111111111`

Reply:
Agreed, these are updated.